cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowToken.java

```diff
@@ -153,32 +183,53 @@ public String get(String key, String nodeName) {
     return null;
   }
 
-  @Nullable
   @Override
-  public String get(String key) {
-    List<NodeValueEntry> nodeValueList = tokenValueMap.get(key);
-    if (nodeValueList == null) {
-      return null;
+  public List<NodeValueEntry> getAll(String key) {
+    return getAll(key, Scope.USER);
+  }
+
+  @Override
+  public List<NodeValueEntry> getAll(String key, Scope scope) {
+    if (tokenValueMap.get(scope).containsKey(key)) {
+      return ImmutableList.copyOf(tokenValueMap.get(scope).get(key));
     }
+    return ImmutableList.of();
+  }
 
-    if (nodeValueList.isEmpty()) {
-      // List of NodeValueEntry cannot be empty if the key is added in the WorkflowToken as
-      // when we add key, we also add single NodeValueEntry.
-      throw new IllegalStateException(String.format("List of NodeValueEntry for the key %s cannot be empty", key));
+  @Override
+  public Map<String, Value> getAllFromNode(String nodeName) {
+    return getAllFromNode(nodeName, Scope.USER);
+  }
+
+  @Override
+  public Map<String, Value> getAllFromNode(String nodeName, Scope scope) {
+    Map<String, Value> tokenValuesFromNode = Maps.newHashMap();
+    for (Map.Entry<String, List<NodeValueEntry>> entry : tokenValueMap.get(scope).entrySet()) {
```

Comment:
``tokenValueMap.get(scope)`` could be null here. However,  doing https://github.com/caskdata/cdap/pull/3128/files#r34424148 will fix it.